nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/RecordReader.java

```diff
@@ -38,14 +38,35 @@
 public interface RecordReader extends Closeable {
 
     /**
-     * Returns the next record in the stream or <code>null</code> if no more records are available.
+     * Returns the next record in the stream or <code>null</code> if no more records are available. Schema enforcement will be enabled.
      *
      * @return the next record in the stream or <code>null</code> if no more records are available.
      *
      * @throws IOException if unable to read from the underlying data
      * @throws MalformedRecordException if an unrecoverable failure occurs when trying to parse a record
+     * @throws SchemaValidationException if a Record contains a field that violates the schema and cannot be coerced into the appropriate field type.
      */
-    Record nextRecord() throws IOException, MalformedRecordException;
+    default Record nextRecord() throws IOException, MalformedRecordException {
```

Comment:
we should indicate whether the scheme enforcement strictness is 'lenient' or 'strict'.